### ISSUE TITLE
Private sharing

### ISSUE THREAD

**Issue by stevenbird** [aikuma/aikuma#428]
Extend `GoogleDriveStorage.gapi_share_with()` to support sharing between two users' private google drives.

**Comment by hleeldc** [comment]
This should be already working as central google drive is just one of the private google drives.

When initialized, GoogleDriveStorage can take the user's email address instead of the server account's email (lp20.org@gmail.com). Then, the shared files will be directed to the user's shared file folder.

```java
String emailAddr = "john@gmail.com";
String identifier = "aikuma/v01/t/te/test@gmail.com/items/ualwfcixlzpu/ualwfcixlzpu-testrnrmf@gmail.com-source.wav";
DataStore ds = new GoogleDriveStorage(accessTooken, rootId, emailAddr);
ds.share(identifier);
```

**Comment by stevenbird** [comment]
New use case: UserA privately shares `File` with UserB from Device1. UserA subsequently adds `Tag` to `File` from Device2. This tag needs to be shared also. Therefore we store sharing data as a tag, in the `/admin` folder, cf #537.

**Comment by stevenbird** [comment]
Resolved by #542
